### PR TITLE
✨New package `react-server`

### PR DESCRIPTION
**Pull request by cartogram**
## Description

This PR adds a new package for providing a server that uses our `react-html` and `react-effect`, giving both rails and node apps a turn-key SSR solution.

I will do a documentation update once the general API is approved.

### CODE REVIEW

File: packages/react-self-serializers/README.md [hunk 1]
```diff
@@ -92,3 +92,15 @@ function App({locale}: {locale?: string}) {
   return <I18n locale={locale}>{/* rest of the app */}</I18n>;
 }
 ```
+
```
Comment (cartogram) [inline packages/react-self-serializers/README.md]: this got onto the wrong branch, ignore

File: packages/react-server/src/render/render.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    applyToContext(ctx, networkManager);
+
+    const immediateAsyncAssets = asyncAssetManager.used(AssetTiming.Immediate);
+    const [styles, scripts] = await Promise.all([
+      assets.styles({name: 'main', asyncAssets: immediateAsyncAssets}),
+      assets.scripts({name: 'main', asyncAssets: immediateAsyncAssets}),
+    ]);
+
+    const response = renderMarkup(
+      <Html locale="fr" manager={htmlManager} styles={styles} scripts={scripts}>
```
Comment (TheMallen) [inline packages/react-server/src/render/render.tsx]: @GoodForOneFare  the version of this in [the rails proving ground](https://github.com/Shopify/web-rails-proving-ground/blob/master/packages/%40shopify/react-server/render/render.tsx#L80) is more up-to-date and has the locale functionality completed.

File: packages/react-server/README.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
+# `@shopify/react-server`
```
Comment (TheMallen) [inline packages/react-server/README.md]: I still like "Loom" :P

File: packages/react-server/src/render/render.tsx (hunk 2 above)
Comment (GoodForOneFare) [reply packages/react-server/src/render/render.tsx]: Ah, cool cool.  I'll fold that into this PR. Thanks! 😸

**Pull request merged**